Revit_Core_Engine/Query/ParameterGroupTypeId.cs

```diff
@@ -0,0 +1,95 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2025, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using System.ComponentModel;
+using BH.oM.Base.Attributes;
+#if !(REVIT2020 || REVIT2021 || REVIT2022)
+using System.Collections.Generic;
+using System.Reflection;
+using System;
+using System.Linq;
+#endif
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Query
+    {
+#if !(REVIT2020 || REVIT2021 || REVIT2022)
+        /***************************************************/
+        /****              Private fields               ****/
+        /***************************************************/
+
+        private static Dictionary<string, ForgeTypeId> m_GroupTypeIdByName = new Dictionary<string, ForgeTypeId>();
+
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Gets the GroupTypeId of a Revit parameter group by its name in the UI.")]
+        [Input("groupName", "Name in the UI of a Revit parameter group.")]
+        [Output("groupTypeId", "GroupTypeId of a Revit parameter group by its name in the UI.")]
+        public static ForgeTypeId ParameterGroupTypeId(this string groupName)
```

Comment:
I had a closer look at this method, I would consider migrating the code out of `Create.Parameter` as an overload for Revit 2020-24 returning `BuiltInParameterGroup`, then calling `ParameterGroupTypeId` for all versions of Revit inside `Create.Parameter`. Hopefully I did not confuse it too much 😉 Let's pick it up tomorrow on the catchup if unclear or not working 👍

Reply:
Yes, exactly!